test/src/test/java/hudson/model/queue/MaintainCanTakeStrengthening.java

```diff
@@ -0,0 +1,78 @@
+package hudson.model.queue;
+
+import hudson.model.FreeStyleProject;
+import hudson.model.Messages;
+import hudson.model.Node;
+import hudson.model.Queue;
+import hudson.model.labels.LabelExpression;
+import hudson.slaves.DumbSlave;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.LoggerRule;
+import org.jvnet.hudson.test.TestExtension;
+
+import java.util.logging.Level;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasItem;
+
+public class MaintainCanTakeStrengthening {
+    @Rule
+    public JenkinsRule r = new JenkinsRule();
+
+    @Rule
+    public LoggerRule logging = new LoggerRule().record(Node.class.getName(), Level.ALL).capture(100);
+
+    private QueueTaskFuture scheduleBuild(String name, String label) throws Exception {
+        FreeStyleProject project = r.createFreeStyleProject(name);
+
+        project.setAssignedLabel(LabelExpression.get(label));
+        return project.scheduleBuild2(0);
+    }
+
+    @Issue("JENKINS-59886")
+    @Test
+    public void testExceptionOnNodeProperty() throws Exception {
+        // A node throwing the exception because of the canTake method of the attached FaultyNodeProperty
+        DumbSlave faultyAgent = r.createOnlineSlave(LabelExpression.get("faulty"));
+        faultyAgent.getNodeProperties().add(new FaultyNodeProperty());
+
+        // A good agent
+        r.createOnlineSlave(LabelExpression.get("good"));
+
+        // Only the good ones will be run and the latest doesn't get hung because of the second
+        QueueTaskFuture[] taskFuture = new QueueTaskFuture[3];
+        taskFuture[0] = scheduleBuild("good1", "good");
+        taskFuture[1] = scheduleBuild("theFaultyOne", "faulty");
+        taskFuture[2] = scheduleBuild("good2", "good");
+
+        // Wait until the good ones are completed
+        while (r.getInstance().getQueue().getLeftItems().size() < 2) {
```

Comment:
Perhaps better to just use taskFuture[0].getStartCondition().get(...) and taskFuture[2].getStartCondition().get(...)?

I don't think you need to call maintain() if things are working right.

Reply:
Ok, I did it to do the test faster, but after measuring, the test is faster without calling `maintain`. Also no need to wait for the completion of the build, the start is proof that the fix works, actually. Thanks